testing/scenario_app/android/app/src/main/java/dev/flutter/scenarios/TestActivity.java

```diff
@@ -70,13 +68,10 @@ public FlutterShellArgs getFlutterShellArgs() {
   @Override
   public void onFlutterUiDisplayed() {
     final Intent launchIntent = getIntent();
-    if (!launchIntent.hasExtra("scenario")) {
-      return;
-    }
     MethodChannel channel =
         new MethodChannel(getFlutterEngine().getDartExecutor(), "driver", JSONMethodCodec.INSTANCE);
     Map<String, Object> test = new HashMap<>(2);
-    test.put("name", launchIntent.getStringExtra("scenario"));
+    test.put("name", "animated_color_square");
```

Comment:
is this right? Seems like this is the mechanism to control which test to run no?

Reply:
+1

Reply:
The launch intent we get here is an integer, not a string. The method channel used expects a string with the name of the test.

Reply:
`launchIntent.getStringExtra("scenario")` should equal to the values defined in https://github.com/flutter/engine/blob/2b11d093e689647c8ad1419a253219b3cd7632c1/testing/scenario_app/android/app/src/androidTest/java/dev/flutter/scenariosui/PlatformViewUiTests.java#L36

Reply:
We should use a different extra name though - because firebase testlab sets this as an integer.

Reply:
is "scenario" a reserved word by Firebase?

Reply:
For gameloops yes.

Reply:
oh lol. I was wondering why we never caught this. Renaming sg. We need to do it everywhere in this project to keep things working.

Reply:
Done - renamed the intent extra to `scenario_name` for these and use it if it's available.

Reply:
Awesome! thanks